### ISSUE TITLE
Problems capturing Canvas Text

### ISSUE THREAD

**Issue by torstah** [yasirkula/Unity360ScreenshotCapture#1]
Hi, 

thanks for a great plugin! Having issues capturing GUI canvas elements such as text. Is that supported? 

Thanks!

**Comment by yasirkula** [comment]
I don't think Unity's RenderToCubemap function (which is used to capture the 360° image) supports it:

- https://feedback.unity3d.com/suggestions/support-for-world-space-ui-elements-in-rendertocubemap
- https://issuetracker.unity3d.com/issues/rendertextures-do-not-include-canvas-ui-elements

**Comment by torstah** [comment]
Ok thanks. Any hints on alternatives?

**Comment by yasirkula** [comment]
I've tried creating a cubemap manually and rendering to each of the 6 sides manually via Camera.Render but it was way too slow and it didn't really work either, lol. _Screen Space-Overlay_ canvases just don't get rendered and _Screen Space-Camera_ canvases get rendered on each face like a duplicate. Only _World Space_ canvases work but then, some objects in the scene started to disappear from my output image.

In conclusion, this is not really possible with just Camera.Render or Camera.RenderToCubemap. It will probably require very complex operations like rendering the UI alone to a texture and then assigning that texture to a temporary quad in front of the camera; but I don't even know if it will work well with UI elements outside of the camera's frustum.

**Comment by torstah** [comment]
I understand, thanks for investigating further!

**Comment by Helaly96** [comment]
Anyways,
i added this script to the camera:
https://forum.unity.com/threads/support-for-ui-elements-in-rendertocubemap.429201/ (Comment 4)
and it works now.
